### PR TITLE
Improve CMake build

### PR DESCRIPTION
**Pull request by joggee-fr**
Hi,

I consider adding a recipe in Yocto project to integrate WiringPi, possibly back in [meta-raspberrypi](https://github.com/agherzan/meta-raspberrypi). The CMake build really seems to be the simplest way to do so.
Consequently, I made some modifications in, especially for dependency management and shared library version.

### CODE REVIEW

**Review by mwallner — APPROVED** [review]

File: devLib/CMakeLists.txt [hunk 1]
```diff
@@ -22,13 +22,16 @@ set(HEADERS
 )
 
 # Compiler flags
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wformat=2 -Winline -pipe -fPIC")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wformat=2 -Winline -pipe")
```
Comment (mwallner) [inline devLib/CMakeLists.txt:25]: why did you remove the PIC flag?

Comment (joggee-fr) [reply devLib/CMakeLists.txt:25]: Described in commit [82ea43eb522f0d0fa8ca0a7c5d6eed9a1f40664f](https://github.com/WiringPi/WiringPi/pull/269/commits/82ea43eb522f0d0fa8ca0a7c5d6eed9a1f40664f), this is default behavior for CMake to build position independent code for shared libraries : https://cmake.org/cmake/help/v3.10/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE.
Otherwise, we don't have to manually set CFLAGS for gcc but would be more portable to set the POSITION_INDEPENDENT_CODE target property in CMake.

Comment (mwallner) [reply devLib/CMakeLists.txt:25]: alright, thanks for clarification!
We'll get this merged now for release 3.9 (internal) and subsequently for 3.10 (the next public after 3.8 - which is due today/tomorrow and we don't have time to re-test everything with this change)
